tests/org.eclipse.n4js.xpect.tests/model/validation/dataflow/NullUndefined_May_OptionalChaining.n4js.xt

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2017 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+
+/*	XPECT_SETUP org.eclipse.n4js.xpect.tests.N4JSXpectTest 
+	IssueConfiguration {
+		IssueCode "DFG_NULL_DEREFERENCE" {enabled=true}
+	}
+	END_SETUP
+ */
+
+class C {
+	field: C;
+}
```

Comment:
You might wanna add a test case for the "null not passed through optional chaining" rule:
```javascript
function f() {
	let c : C = new C();
	if (true) {
		c = null;
	}
	// XPECT warnings --> "Variable c may be null" at "x"
	c.field;

	let x = c?.field;
	// XPECT warnings --> "Variable x may be undefined" at "x"
	x.toString();
}
```